lib/wiki_discouraged_article_manager.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+require_dependency "#{Rails.root}/lib/importers/category_importer"
+
+class WikiDiscouragedArticleManager
+  def retrieve_wiki_edu_discouraged_articles
+    category_title = ENV['blocked_assignment_category']
+    discouraged_articles = CategoryImporter.new(Wiki.default_wiki)
+                                           .page_titles_for_category("Category:#{category_title}")
+    update_or_save_discouraged_articles(discouraged_articles, category_title)
+  end
+
+  private
+
+  def update_or_save_discouraged_articles(discouraged_articles, category_title)
+    existing_article = Category.find_by(name: category_title)
```

Comment:
This could find the wrong category, because `name` is not a unique identifier for a Category record.  The combination of name, wiki, source, and depth is unique.

I suggest using `Category.find_or_create_by`, which will let you simplify the rest of this method.